### DISCUSSION TITLE
v1.8.0

### DISCUSSION THREAD

**Discussion by dom-robinson** [jekhokie/raspberry-noaa-v2#459] [General]
# Raspberry-NOAA-V2 (‘RN2’) - Release 1.8.0 

## ‘Defaulting to gnuradio’
A number of users have expressed an interested to support a wider variety of SDR hardware devices beyond those interfaced to the original default rtl-sdr cards through the rtl-fm based commands used in the associated drivers.

In the original incarnation of RN2 the NOAA captures were driven directly through the [audio_processor ](https://github.com/jekhokie/raspberry-noaa-v2/blob/master/scripts/audio_processors/)scripts to use rtl_fm directly, and the METEOR-M2 captures were driven using a python script that was originally generated in gnuradio and used the OSMOCOM source block, which in turn uses the rtl_fm binary to interface to the SDR card.

Since gnuradio can allow interchange of such ‘source blocks’ this methodology potentially opens up RN2 to support wider SDR hardware options.

This release DOES NOT seek to provide immediate access to the wider hardware options, but it DOES allow the community to begin to interface their own hardware by modifying the gnuradio composer files (.grc files included in the /audio_processors/ folder, to export python (.py) scripts similar to those included for the SDR, to copy the modifications that must then be made manually to the .py to interface the python workflow to the RN2 workflow (documented in the examples provided) and to then use alternative hardware to capture the source .wav or .s / .qpsk files that then can be used in the /image_processors/ scripts.

On day one most users defaulting to ‘gnuradio’ in their settings.yml will see little difference to RN2, but over time the community can now look to a common method to introduce new hardware support to the project.

More details on the deeper code is included in the new [README](https://github.com/jekhokie/raspberry-noaa-v2/blob/master/scripts/audio_processors/README.md) included in the audio_processors folder, which includes sample .grc files and annotation in the current 'default' .py files showing where edits must be manually made AFTER exporting the python from gnuradio composer to make the scripts work with the RN2 settings.

## Cosmetics
The spectrogram has been moved from SOX generation to FFMPEG generation, and also is represented vertically instead of horizontally. In the next iteration it may be suitable to flip it along with the NORTHBOUND passes so that as displayed in webpanel the timeline will alway flow in the same direction as the pass images shown (interference bars will then correlate in parallel between the two). It also means that colour-schema and other elements in the spectrograms can be more customised by those who want to do so.

## Enhancements
We have enabled as many enhancements as possible in the NOAA workflows, and the webpanel now includes a hyperlink to a ‘details’ page that contains much of the documentation from the wxtoimg command line manual that relates to each enhancement type. The list shows only those that there are scripts in the /image_processors/ folder - there are a couple purposefully left out, but would be easy to introduce for those keen to do so. 

## Kudos
- @colinluthier  for his continued support, critical eye and key work on various elements of the ‘plumbing’
- @Cadair for git fu, many general code contributions and for sanity checks on a nearly daily basis!

@everyone for testing and being patient!

## What's Changed
* Add Missing Config to Schema and ffmpeg by @jekhokie in https://github.com/jekhokie/raspberry-noaa-v2/pull/337
* Add Advanced Scheduling by @jekhokie in https://github.com/jekhokie/raspberry-noaa-v2/pull/340
* script link incorrect by @b3rtdb in https://github.com/jekhokie/raspberry-noaa-v2/pull/376
* Auto Gain fix by @cdtush in https://github.com/jekhokie/raspberry-noaa-v2/pull/386
* Empty image bug patch by @colinluthier in https://github.com/jekhokie/raspberry-noaa-v2/pull/363
* Update push scripts so output is logged to the file by @Cadair in https://github.com/jekhokie/raspberry-noaa-v2/pull/387
* METEOR on TMPFS depending on free memory by @Clnmt in https://github.com/jekhokie/raspberry-noaa-v2/pull/359
* Prune older than patch by @colinluthier in https://github.com/jekhokie/raspberry-noaa-v2/pull/373
* Add a push provider to Matrix by @Cadair in https://github.com/jekhokie/raspberry-noaa-v2/pull/362
* put back missing space by @Cadair in https://github.com/jekhokie/raspberry-noaa-v2/pull/391
* French i18n by @Clnmt in https://github.com/jekhokie/raspberry-noaa-v2/pull/389
* Add Slack push feature by @MavEtJu in https://github.com/jekhokie/raspberry-noaa-v2/pull/375
* Fix config validation by @Cadair in https://github.com/jekhokie/raspberry-noaa-v2/pull/392
* FLIP capture only if PASS_DIRECTION is Northbound by @Clnmt in https://github.com/jekhokie/raspberry-noaa-v2/pull/398
* Disable sending of mail after "at" job execution by @Clnmt in https://github.com/jekhokie/raspberry-noaa-v2/pull/403
* Portuguese (PT) version by @airlomba in https://github.com/jekhokie/raspberry-noaa-v2/pull/426
* Support Chinese by @showdoi in https://github.com/jekhokie/raspberry-noaa-v2/pull/443
* Fix #400 by using which to find path for meteor_demod. by @ajcollins in https://github.com/jekhokie/raspberry-noaa-v2/pull/450
* Update receive_meteor.sh to use available memory rather than free memory by @ajcollins in https://github.com/jekhokie/raspberry-noaa-v2/pull/451
* Update logged stats in receive_meteor.sh to include available memory. by @ajcollins in https://github.com/jekhokie/raspberry-noaa-v2/pull/452
* Add libsox-fmt-mp3 to dependencies by @Miskerest in https://github.com/jekhokie/raspberry-noaa-v2/pull/453
* Histogram upgrade by @colinluthier in https://github.com/jekhokie/raspberry-noaa-v2/pull/361
* enabling gnuradio recording option for M2 and NOAA by @dom-robinson in https://github.com/jekhokie/raspberry-noaa-v2/pull/437

## New Contributors
* @b3rtdb made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/376
* @cdtush made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/386
* @Cadair made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/387
* @Clnmt made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/359
* @MavEtJu made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/375
* @airlomba made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/426
* @showdoi made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/443
* @ajcollins made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/450
* @Miskerest made their first contribution in https://github.com/jekhokie/raspberry-noaa-v2/pull/453

**Full Changelog**: https://github.com/jekhokie/raspberry-noaa-v2/compare/v1.7.0...v1.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/jekhokie/raspberry-noaa-v2/releases/tag/v1.8.0'>v1.8.0</a>.</em>

**Comment by Cadair** [comment]
Thanks to everyone who contributed to this release :rocket: :artificial_satellite:
